File: src/main/java/me/glorantq/aboe/chat/common/PacketChannelList.java
package me.glorantq.aboe.chat.common;

import cpw.mods.fml.common.network.simpleimpl.IMessage;
import io.netty.buffer.ByteBuf;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import me.glorantq.aboe.chat.client.channels.ClientChatChannel;

import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
public class PacketChannelList implements IMessage {
    private List<ClientChatChannel> channels;

    @Override
    public void fromBytes(ByteBuf buf) {
        this.channels = new ArrayList<>();
        int length = buf.readInt();
        for(int i = 0; i < length; i++) {
            this.channels.add(ClientChatChannel.fromBytes(buf));
        }
    }

    @Override
    public void toBytes(ByteBuf buf) {
        buf.writeInt(channels.size());
        for(ClientChatChannel channel : channels) {
            channel.toBytes(buf);
        }
    }
}

File: src/main/kotlin/me/glorantq/aboe/chat/common/PacketChannelList.kt
package me.glorantq.aboe.chat.common

import cpw.mods.fml.common.network.simpleimpl.IMessage
import io.netty.buffer.ByteBuf
import me.glorantq.aboe.chat.ClientChannelList
import me.glorantq.aboe.chat.client.channels.ClientChatChannel
import java.util.*

data class PacketChannelList(var channels: ClientChannelList) : IMessage {
    constructor() : this(mutableListOf())

    override fun fromBytes(buf: ByteBuf) {
        this.channels = ArrayList()
        val length = buf.readInt()
        for (i in 0 until length) {
            this.channels.add(ClientChatChannel.fromBytes(buf))
        }
    }

    override fun toBytes(buf: ByteBuf) {
        buf.writeInt(channels.size)
        for (channel in channels) {
            channel.toBytes(buf)
        }
    }
}
